chew.go

```diff
@@ -67,6 +212,28 @@ var validExtensions = map[string]func(io.Reader, string) ([]common.Chunk, error)
 	".epub": document.ProcessEpub,
 }
 
+var httpClient *http.Client
+
+// Obviously, this is not the best way to do this but it's a start
+
+/*
+SetHTTPClient allows you to set a custom http.Client to use for making requests.
+
+This would be useful in the event custom logging, tracing, or other functionality is
+required for the requests made by the library.
+
+Usage:
+
+	client := &http.Client{
+		Transport: loggingRoundTripper{wrapped: http.DefaultTransport},
+	}
+
+	chew.SetHTTPClient(client)
+*/
+func SetHTTPClient(client *http.Client) {
+	httpClient = client
+}
```

Comment:
This should be something that is not at package level, but a method of an object

Your NewConfig object is a good idea, but you should have a Chew instance returned, then SettHTTPClient can be used on it

All your existing methods should be replaced to me method of the object

Then if you want to keep a retro compatibility, you should create a defaultChew = NewChewig(common.Config{})

then you rewrite existing exported method to call the defaultChew

```go
func Process(urls []string, ctxs ...context.Context) ([]common.Chunk, error) {
	ctx := context.Background()
	if len(ctxs) > 0 {
		ctx = ctxs[0]
	}
	return defaultChew.Process(ctx, urls)
}
```